spec/consumer_spec.rb

```diff
@@ -0,0 +1,65 @@
+require_relative 'spec_helper'
+
+describe Consumer do
+
+  let(:config) {{
```

Comment:
I've gave this a shot and the problem is that all the `let` blocks are for things inside purely internal to `Consumer` and a chain of things have to happen even before you get to the `queue`.

I definitely think moving the `config` to the spec_helper is a good idea though.